src/osemgrep/cli_ui/Console_Spinner.ml

```diff
@@ -4,18 +4,25 @@
 
 let spinner = [| "⠋"; "⠙"; "⠹"; "⠸"; "⠼"; "⠴"; "⠦"; "⠧"; "⠇"; "⠏" |]
 
+(* only show the console spinner if this is a non-JSOO Unix TTY *)
+let should_show_spinner () =
+  !ANSITerminal.isatty Unix.stdout && (not !Common.jsoo) && Sys.unix
+
 let show_spinner delay_ms : unit =
-  let print_frame ~frame_index:i : unit =
-    let spinner = spinner.(i mod Array.length spinner) in
-    ANSITerminal.set_cursor 1 (-1);
-    ANSITerminal.printf [ ANSITerminal.green ] "%s Waiting for sign in..."
-      spinner
-  in
-  for frame_index = 1 to 100 do
-    print_frame ~frame_index;
-    (* Note: sleep is measured in seconds *)
-    Unix.sleepf (Float.of_int delay_ms /. Float.of_int (1000 * 100))
-  done
+  if not (should_show_spinner ()) then
+    ANSITerminal.printf [ ANSITerminal.green ] "Waiting for sign in..."
```

Comment:
I'd note that this spinner is also intended to introduce a delay between successive requests to our login endpoint.

Reply:
Gotcha, do you think I should add a `Unix.sleepf` of delay_ms * 100 (mimicking the frame_index for loop)?

Reply:
(I'm going to merge for now to unblock myself -- happy to add this in a subsequent PR if you agree)